OpenEdX/Managers/PushNotificationsManager/PushNotificationsManager.swift

```diff
@@ -94,4 +102,58 @@ class PushNotificationsManager {
             listener.didReceiveRemoteNotification(userInfo: userInfo)
         }
     }
+    
+    func synchronizeToken() {
+        for provider in providers {
+            provider.synchronizeToken()
+        }
+    }
+    
+    func refreshToken() {
+        for provider in providers {
+            provider.refreshToken()
+        }
+    }
+}
+
+// MARK: - MessagingDelegate
+extension PushNotificationsManager: MessagingDelegate {
+    func messaging(_ messaging: Messaging, didReceiveRegistrationToken fcmToken: String?) {
+        for provider in providers where provider is MessagingDelegate {
+            (provider as? MessagingDelegate)?.messaging?(messaging, didReceiveRegistrationToken: fcmToken)
+        }
+    }
+}
+
+// MARK: - UNUserNotificationCenterDelegate
+extension PushNotificationsManager: UNUserNotificationCenterDelegate {
+    @MainActor
+    func userNotificationCenter(
+        _ center: UNUserNotificationCenter,
+        willPresent notification: UNNotification
+    ) async -> UNNotificationPresentationOptions {
+        // With swizzling disabled you must let Messaging know about the message, for Analytics
+        Messaging.messaging().appDidReceiveMessage(notification.request.content.userInfo)
```

Comment:
I guess the right place for this is FCMListner.

Reply:
It requires extra abstraction for PushNotificationsProvider; it can create overhead.
For me, it seems ok to leave just 1 line here.
wdyt?

Reply:
https://github.com/openedx/openedx-app-ios/pull/461#discussion_r1646842373

Reply:
https://github.com/openedx/openedx-app-ios/pull/461#discussion_r1647211604